bentoml/cli/__init__.py

```diff
@@ -118,24 +121,26 @@ def cli():
 
     # pylint: disable=unused-variable
 
-    @_cli.command(help='Generate serverless project with BentoML service archive.',
-                  context_settings=dict(ignore_unknown_options=True, allow_extra_args=True))
+    @_cli.command(help='Deploy BentoML archive to AWS Lambda or Google Cloud Function as ' +
+                  'REST endpoint with Serverless Framework', context_settings=dict(
+                      ignore_unknown_options=True, allow_extra_args=True))
     @click.argument('archive-path', type=click.STRING)
-    @click.argument('output-path', type=click.STRING)
-    @click.option('--platform', type=click.Choice(['aws-python', 'aws-python3', 'google-python']),
-                  default='aws-python3')
+    @click.option('--platform', type=click.Choice([
+        'aws-lambda', 'aws-lambda-py3', 'gcp-function', 'aws-sagemaker', 'azure-ml', 'algorithmia'
+    ]), required=True)
     @click.pass_context
-    def build_serverless_archive(ctx, archive_path, output_path, platform):
-        if which('serverless') is None:
-            click.echo('Serverless framework is not installed', err=True)
-            click.echo('Please visit www.serverless.com for install instructions')
+    def deploy(ctx, archive_path, platform):
+        if platform in SERVERLESS_PLATFORMS:
+            bento_service = load(archive_path)
+            output_path = deploy_with_serverless(bento_service, platform, archive_path, ctx.args)
```

Comment:
might be better to encapsulate all related logics into the `deploy_with_serverless` call, that way we can reuse it in another context more easily.

```
if platform in SERVERLESS_PLATFORMS:
  extra_args = default_serverless_parser.parse_args(ctx.args)
  saved_path  = deploy_with_serverless(archive_path, platform, extra_args)
  click.echo(...)
```

I am thinking we can provide a python API later on, so that uses can do this from their python environment directly. e.g.:
```
saved_path = my_svc.pack(model=model)
bentoml.deploy_to_aws_lambda(saved_path, { region: 'us-west-2' })
```

Reply:
just realized, it might make more sense to move the serverless.py file to 'bentoml/deployment/serverless.py'